build.gradle.kts

```diff
@@ -0,0 +1,150 @@
+}
+
+tasks {
+    processResources {
+        filteringCharset = "UTF-8"
+        filesMatching("plugin.yml") {
+            expand("version" to version)
+        }
+    }
+
+    jar {
+        finalizedBy("shadowJar")
+    }
+    shadowJar {
+        mustRunAfter("build")
+
+        relocate("net.dv8tion.jda", "github.scarsz.discordsrv.dependencies.jda")
+        relocate("com.iwebpp.crypto", "github.scarsz.discordsrv.dependencies.iwebpp.crypto")
+        relocate("com.vdurmont.emoji", "github.scarsz.discordsrv.dependencies.emoji")
+        relocate("com.neovisionaries.ws", "github.scarsz.discordsrv.dependencies.ws")
+        relocate("net.kyori", "github.scarsz.discordsrv.dependencies.kyori")
+        relocate("dev.vankka.mcdiscordreserializer", "github.scarsz.discordsrv.dependencies.mcdiscordreserializer")
+        relocate("dev.vankka.simpleast", "github.scarsz.discordsrv.dependencies.simpleast")
+        relocate("org.bstats", "github.scarsz.discordsrv.dependencies.bstats")
+        relocate("org.apache.commons", "github.scarsz.discordsrv.dependencies.commons")
+        relocate("org.json.simple", "github.scarsz.discordsrv.dependencies.json.simple")
+        relocate("org.json", "github.scarsz.discordsrv.dependencies.json")
+        relocate("org.minidns", "github.scarsz.discordsrv.dependencies.minidns")
+        relocate("okhttp3", "github.scarsz.discordsrv.dependencies.jda")
+        relocate("okio", "github.scarsz.discordsrv.dependencies.okio")
+        relocate("gnu.trove", "github.scarsz.discordsrv.dependencies.trove")
+        relocate("com.fasterxml.jackson", "github.scarsz.discordsrv.dependencies.jackson")
+        relocate("com.google.common", "github.scarsz.discordsrv.dependencies.google.common")
+        relocate("com.google.errorprone", "github.scarsz.discordsrv.dependencies.google.errorprone")
+        relocate("com.google.gson", "github.scarsz.discordsrv.dependencies.google.gson")
+        relocate("com.google.j2objc", "github.scarsz.discordsrv.dependencies.google.j2objc")
+        relocate("com.github.kevinsawicki", "github.scarsz.discordsrv.dependencies.kevinsawicki")
+        relocate("com.github.zafarkhaja", "github.scarsz.discordsrv.dependencies.zafarkhaja")
+        relocate("alexh", "github.scarsz.discordsrv.dependencies.alexh")
+
+        // Merge META-INF/services files where needed
+        mergeServiceFiles()
+
+        // Exclude signatures, maven/ and proguard/ from META-INF
+        exclude("META-INF/*.SF")
+        exclude("META-INF/*.DSA")
+        exclude("META-INF/*.RSA")
+        exclude("META-INF/maven/**")
+        exclude("META-INF/proguard/**")
+    }
+}
+
+repositories {
+    mavenLocal()
+    mavenCentral()
+    maven("https://hub.spigotmc.org/nexus/content/repositories/snapshots/")
+    maven("https://papermc.io/repo/repository/maven-public/")
+    maven("https://oss.sonatype.org/content/repositories/snapshots")
+    maven("https://nexus.scarsz.me/content/groups/public/")
+}
+
+dependencies {
```

Comment:
On the topic of dependency configurations,

- `adventure-api` needs to be `api` since Components are exposed in events, MessageUtil
  - adventure serializers, minimessage and mcdiscordreserializer should be `api`
- `configuralize` should probably be `api`, somebody is probably getting something from there
- `org.jetbrains:annotations` should probably be `compileOnlyApi` to avoid people getting a bunch of errors in decompiled files

Not sure where we should draw the line for the rest of the (internal-ish) dependencies, or if we should just make most dependencies api. I know that [Staff-Chat](https://github.com/DiscordSRV/Staff-Chat/blob/92d814b5835270ff54d7c4947949b9468d7145df/src/main/java/com/rezzedup/discordsrv/staffchat/MessageProcessor.java#L33) uses emoji-java